ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,100 @@
+# 39. Workspace Deprecation
+
+Some of the text was taken from - https://github.com/konflux-ci/architecture/pull/187
+
+* Date: 2024-09-22
+
+## Status
+
+Accepted
+
+## Context
```

Comment:
Can we link this ADR with previous ones that are now superseded? i.e. `31. Sprayproxy`